File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.cpp
class Solution {
public:
    vector<int> applyOperations(vector<int>& nums) {
        int ctZero=0;
        for(int i=0; i<nums.size()-1; i++)
            if(nums[i]==nums[i+1])
                nums[i]*=2, nums[i+1]=0;
        // for(int lt=0,rt=1; max(lt,rt)<nums.size();){
        //     if (nums[lt])   // find 0s
        //         lt++;
        //     while(rt<=lt)
        //         rt++;
        //     if (!nums[rt])  // find non-zeros
        //         rt++;
        //     else swap(nums[lt], nums[rt]);
        // }
        for(int lt=0,rt=0; rt<nums.size();rt++){
            if(nums[rt])
                nums[lt++]=nums[rt];
            else ctZero++;
        }
        int rt=nums.size()-1;
        while(ctZero--)
            nums[rt--]=0;
        return nums;
    }
};

File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.java
class Solution {
    public int[] applyOperations(int[] nums) {
        int n = nums.length, lt = 0, rt = 0;
        for (; rt < n; rt++) {
            if (nums[rt] == 0)
                continue;
            if (rt < n - 1 && nums[rt] == nums[rt + 1]) {
                nums[lt] = nums[rt] * 2;
                lt++;
                rt++;
            } else {
                nums[lt] = nums[rt];
                lt++;
            }
        }
        while (lt < n)
            nums[lt++] = 0;
        return nums;
    }
}